### PR TITLE
Add test coverage for GUID utilities.

### PR DESCRIPTION
**Pull request by hidmic**
Precisely what the title says. These tests simply verify that conversion in both directions is possible.

### CODE REVIEW

**Comment by hidmic** [comment]
CI up to `rmw_fastrtps_shared_cpp`:


* Coverage [Build Status](https://ci.ros2.org/job/ci_linux_coverage/99/)

File: rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#include "rmw_fastrtps_shared_cpp/guid_utils.hpp"
+
+using rmw_fastrtps_shared_cpp::copy_from_byte_array_to_fastrtps_guid;
+using rmw_fastrtps_shared_cpp::copy_from_fastrtps_guid_to_byte_array;
+
+static constexpr size_t byte_array_size =
+  eprosima::fastrtps::rtps::GuidPrefix_t::size +
+  eprosima::fastrtps::rtps::EntityId_t::size;
+
+TEST(GUIDUtilsTest, byte_array_to_guid_and_back) {
```
Comment (brawner) [inline rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp:28]: What do you think about checking for null arrays using EXPECT_DEATH()?

File: rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  uint8_t input_byte_array[byte_array_size] = {0};
+  input_byte_array[0] = 0xA5;
+  input_byte_array[byte_array_size - 1] = 0x4B;
+  eprosima::fastrtps::rtps::GUID_t guid;
+  copy_from_byte_array_to_fastrtps_guid(input_byte_array, &guid);
```
Comment (brawner) [inline rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp:33]: While converting back and forth between the two does ensure that no data is lost, it doesn't really test that it was converted to the correct output. It is also possible that this won't catch errors if they are introduced in the header if somehow both methods correct each other's mistake.

File: rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#include "gtest/gtest.h"
+
+#include "fastrtps/rtps/common/Guid.h"
+
+#include "rmw_fastrtps_shared_cpp/guid_utils.hpp"
+
+using rmw_fastrtps_shared_cpp::copy_from_byte_array_to_fastrtps_guid;
+using rmw_fastrtps_shared_cpp::copy_from_fastrtps_guid_to_byte_array;
+
+static constexpr size_t byte_array_size =
```
Comment (brawner) [inline rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp:24]: Out of curiosity, what does byte_array_size evaluate to?

Comment (hidmic) [reply rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp:24]: It's 16, but I preferred not to hard-code it.

File: rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp (hunk 1 above)
Comment (hidmic) [reply rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp:28]: Good point.

File: rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp (hunk 2 above)
Comment (hidmic) [reply rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp:33]: That is true, I just didn't want to pollute the test with too much information as to how it gets copied in-n-out.

File: rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp (hunk 1 above)
Comment (hidmic) [reply rmw_fastrtps_shared_cpp/test/test_guid_utils.cpp:37]: See https://github.com/ros2/rmw_fastrtps/pull/387/commits/7bcd43457d79a0a3108f25c0d26f15ef604f640b

**Comment by brawner** [comment]
Windows failed the test because it didn't output a message when they died, I think it would be fine if the expected matcher was just an empty string.

**Comment by Blast545** [comment]
Why isn't the output available in Windows as in the other platforms though? AFAIK the `assert` also outputs diagnostic information in Windows, should this be investigated?

> I think it would be fine if the expected matcher was just an empty string.

+1

**Comment by hidmic** [comment]
@brawner PTAL

**Pull request merged**